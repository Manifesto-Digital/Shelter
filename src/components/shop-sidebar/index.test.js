import React from 'react';
import {
  createFactory,
  createChildContentfulRichText,
} from '../../utils/test-factories';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../__tests__/helpers/index';
import ShopSidebar from './index';

export const createShopSidebar = createFactory({
  type: 'Boutique',
  displayAddress: createChildContentfulRichText(),
  contactNumber: '01234567890',
  contactEmail: 'email@shelter.org.uk',
  openingHours: createChildContentfulRichText(),
  parkingInformation: createChildContentfulRichText(),
  disabledAccessInformation: createChildContentfulRichText(),
});

it('renders correctly', () => {
  const mockData = createShopSidebar();

  snapshotComponent(<ShopSidebar data={mockData} />);
});

it('displays the Furniture accepted section if the shop is a furniture shop', () => {
  const mockData = createShopSidebar({ type: 'Furniture' });
  const wrapper = mountWithTheme(<ShopSidebar data={mockData} />);

  expect(
    wrapper
      .find('h3')
      .at(3)
      .text()
  ).toEqual('Furniture accepted');
});
