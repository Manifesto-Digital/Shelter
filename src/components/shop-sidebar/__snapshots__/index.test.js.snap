// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 ul {
  padding-left: 0;
  list-style: none;
}

.c1 ul li {
  position: relative;
  padding-left: 40px;
}

.c1 ul li::before {
  content: ' ';
  position: absolute;
  top: 8px;
  left: 10px;
  display: block;
  width: 10px;
  height: 10px;
  background: #212122;
}

.c1 ul li ul {
  padding-left: 0;
  list-style: none;
}

.c1 ul li ul li:before {
  background: transparent;
  border: 1px solid #212122;
}

.c1 ol li::before {
  min-width: 10px;
}

.c0 {
  padding: 20px;
  background-color: #e5e5e5;
}

<div
  className="c0"
>
  <h3>
    Address
  </h3>
  <div
    className="c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Our advice and support services across the UK give people <b>one-to-one</p>",
      }
    }
  />
  <h3>
    Parking
  </h3>
  <div
    className="c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Our advice and support services across the UK give people <b>one-to-one</p>",
      }
    }
  />
  <h3>
    Disabled access
  </h3>
  <div
    className="c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Our advice and support services across the UK give people <b>one-to-one</p>",
      }
    }
  />
  <h3>
    Opening times
  </h3>
  <div
    className="c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Our advice and support services across the UK give people <b>one-to-one</p>",
      }
    }
  />
  <h3>
    Phone
  </h3>
  <p>
    <a
      rel="noopener"
      target="_self"
    >
      01234567890
    </a>
  </p>
  <h3>
    Email
  </h3>
  <p>
    <a
      rel="noopener"
      target="_self"
    >
      email@shelter.org.uk
    </a>
  </p>
</div>
`;
